templates/types/streaming/nextjs/app/components/ui/chat/chat-message/chat-events.tsx

```diff
@@ -8,6 +8,33 @@ import {
 } from "../../collapsible";
 import { EventData } from "../index";
 
+type MergedEvent = {
+  agent?: string;
+  titles: string[];
+};
+
+function mergeAdjacentEvents(events: EventData[]): MergedEvent[] {
+  const mergedEvents: MergedEvent[] = [];
+
+  for (const event of events) {
+    const lastMergedEvent = mergedEvents[mergedEvents.length - 1];
+
+    if (
+      lastMergedEvent &&
+      lastMergedEvent.agent === event.agent &&
+      event.agent !== null
+    ) {
+      // If the last event in mergedEvents has the same non-null agent, add the title to it
+      lastMergedEvent.titles.push(event.title);
+    } else {
+      // Otherwise, create a new merged event
+      mergedEvents.push({ agent: event.agent, titles: [event.title] });
+    }
+  }
+
+  return mergedEvents;
+}
+
 export function ChatEvents({
```

Comment:
we should use a new component for rendering the agent events - the old one we keep the same

Reply:
i would put it in a new file like `chat-agent-events.tsx`